http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/AppComponent.scala

```diff
@@ -151,7 +155,7 @@ object appQuery extends TableQuery(new AppTable(_)) {
           )(DBIO.successful(_))
         )
 
-      // here, we enforce uniqueness on (AppName, GoogleProject) for active apps
+      // here, we enforce uniqueness on (AppName, CloudContext) for active apps
```

Comment:
This logic means that Azure app names must be unique within an entire MRG (not just unique with a workspace). I'm not actually sure if this a requirement in our AKS deployment, but am keeping the logic in place for GCP and Azure.

1 reason I don't like our app data model: these kinds of constraints aren't expressible in the DB and thus enforced in code.

Reply:
this was probably fine for v1 route...maybe we should use sth different for v2? so that the uniqueness is at workspace level?

Reply:
I thought about that -- the k8s deployments do name things like `{{ appName }}-something` so I wasn't sure if there would be name conflicts if we allow duplicate appNames within a cluster. Technically everything is scoped to the namespace in k8s so maybe it's ok. But was thinking this is safer (and the UI generates unique appNames anyway)..

Reply:
Yeah it is a gotcha.. I can try changing v2 apps to be unique by (name, workspace) and see what happens

Reply:
[done](https://github.com/DataBiosphere/leonardo/pull/3024/commits/3c83fe867044e4d52c280c3060ded19a21800529)

Reply:
I successfully created 2 apps with the same name in different workspaces 🎉